Check values in arrays

jsoriano opened the pull request:
Add validation for all values in arrays.

This is needed when some validation of the values is desired, apart of type validation. This is the case with IPs (https://github.com/elastic/integrations/issues/3408) or event types (https://github.com/elastic/elastic-package/issues/615#issuecomment-1021884406).

This change was part of https://github.com/elastic/elastic-package/pull/771, but this is being delayed due to https://github.com/elastic/integrations/issues/3016.

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package%2FPR-835/detail/PR-835/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package%2FPR-835/detail/PR-835/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package%2FPR-835/detail/PR-835/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package%2FPR-835/detail/PR-835/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://elastic-package_835.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2022-05-30T10:17:03.383Z&rangeTo=2022-05-30T10:37:03.383Z&transactionName=Ingest-manager/elastic-package/PR-{number}&transactionType=job&latencyAggregationType=avg&traceId=ea2548de14ebec73d18ae5455d90338c&transactionId=8e32e267946a8bce)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2022-05-30T10:27:03.383+0000


* Duration: 32 min 58 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 720  |
| Skipped      | 0 |
| Total        | 720   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->












## :robot: GitHub comments

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`30/30`) | :green_heart: 
Files | 64.22% (`70/109`) | :+1: 
Classes | 58.278% (`88/151`) | :+1: 
Methods | 47.333% (`284/600`) | :+1: 
Lines | 32.39% (`2620/8089`) | :+1: 0.041
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

jsoriano commented:
> A more generic non-blocking comment - it's really easy to get lost in the logic due to multiple switch-cases and "if" clauses. Maybe you should drop a few comments to show field examples including edge cases.

I have added comments, and refactored the code to explicitly return type errors instead of having the `if valid` condition at the end. I have also moved the code to a function for single values, what reduces the nesting level of the main code.

@mtojek let me know what you think.

mtojek reviewed (approved):
Thank you! That's exactly what I meant.
